packages/uniforms/__tests__/BaseForm.tsx

```diff
@@ -1,387 +1,76 @@
-      expect(context2).toHaveProperty('changed', true);
-      expect(context2).toHaveProperty('changedMap.$');
-      expect(context2.changedMap.$).toBeTruthy();
-      expect(context2).toHaveProperty('changedMap.$.1');
-      expect(context2.changedMap.$?.[1]).toBeTruthy();
-
-      wrapper.instance().getContext().onChange('$', [1]);
-
-      const context3 = wrapper.instance().getContext();
-      expect(context3).toHaveProperty('changed', true);
-      expect(context3).toHaveProperty('changedMap.$');
-      expect(context3.changedMap.$).toBeTruthy();
-      expect(context3).toHaveProperty('changedMap.$.1');
-      expect(context3.changedMap.$?.[1]).toBeTruthy();
-    });
-
-    it('autosaves correctly (`autosave` = true)', async () => {
-      wrapper.setProps({ autosave: true });
-      wrapper.instance().getContext().onChange('a', 1);
-      await new Promise(resolve => setTimeout(resolve));
-      const context = wrapper.instance().getContext();
-      expect(onSubmit).toHaveBeenCalledTimes(1);
-      expect(context.submitted).toBe(true);
-      expect(onSubmit).toHaveBeenLastCalledWith(model);
-    });
-
-    it('autosaves are not delayed', async () => {
-      wrapper.setProps({ autosave: true });
-      wrapper.instance().getContext().onChange('a', 1);
-      await new Promise(resolve => setTimeout(resolve));
-
-      expect(onSubmit).toHaveBeenCalledTimes(1);
-      expect(onSubmit).toHaveBeenLastCalledWith(model);
-    });
-
-    it('autosaves can be delayed', async () => {
-      wrapper.setProps({ autosave: true, autosaveDelay: 25 });
-      wrapper.instance().getContext().onChange('a', 1);
-      wrapper.instance().getContext().onChange('a', 2);
-      wrapper.instance().getContext().onChange('a', 3);
-      await new Promise(resolve => setTimeout(resolve));
-
-      expect(onSubmit).not.toHaveBeenCalled();
-
-      await new Promise(resolve => setTimeout(resolve, 25));
-
-      expect(onSubmit).toHaveBeenCalledTimes(1);
-      expect(onSubmit).toHaveBeenLastCalledWith(model);
-    });
-
-    it('autosaves can be delayed (longer)', async () => {
-      wrapper.setProps({ autosave: true, autosaveDelay: 10 });
-      wrapper.instance().getContext().onChange('a', 1);
-      wrapper.instance().getContext().onChange('a', 2);
-      wrapper.instance().getContext().onChange('a', 3);
-
-      await new Promise(resolve => setTimeout(resolve, 25));
-
-      wrapper.instance().getContext().onChange('a', 1);
-      wrapper.instance().getContext().onChange('a', 2);
-      wrapper.instance().getContext().onChange('a', 3);
-
-      await new Promise(resolve => setTimeout(resolve, 25));
-
-      expect(onSubmit).toHaveBeenCalledTimes(2);
-      expect(onSubmit).toHaveBeenLastCalledWith(model);
-    });
-
-    it('clears autosave correctly', () => {
-      wrapper.setProps({ autosave: true, autosaveDelay: 100 });
-      wrapper.instance().getContext().onChange('a', 1);
-      wrapper.unmount();
-
-      expect(onSubmit).not.toBeCalled();
-    });
```

Comment:
That one is very important when it comes to the functionality.
https://testing-library.com/docs/react-testing-library/api/#unmount